internal/encode_args.go

```diff
@@ -80,6 +80,11 @@ func decodeArgsToRawValues(dc converter.DataConverter, fnType reflect.Type, data
 	}
 
 	// Unmarshal
+
+	// Dynamic workflows take in a single EncodedValues, encode all data into single EncodedValues
+	if fnType.NumIn() == 2 && len(pointers) == 1 && fnType.In(1) == reflect.TypeOf((*converter.EncodedValues)(nil)).Elem() {
```

Comment:
Hrmm, wonder if it's worth a more specific check whether it's a dynamic workflow/activity instead of assuming so based on signature? Maybe not worth it.

Reply:
Yeah I would consider a more specific check or just not sue this function for dynamic workflows/activities